src/ContentfulImage.js

```diff
@@ -0,0 +1,127 @@
+import React, { forwardRef, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import buildUrlParameters from './utils';
+import { FORMAT, RESIZE, CROP, QUALITY, BACKGROUND_COLOR } from './constants';
+
+const ContentfulImage = forwardRef(({ format, optimize, resize, cropRadius, quality, backgroundColor, ...imageProps }, ref) => {
+    const { convertedUrl, mimeType, originalUrl } = useMemo(() => {
+        const originalUrl = imageProps.src;
+        let urlParameters = '';
+        let returnedMimeType;
+
+        if (optimize) {
+            const { url, mimeType } = buildUrlParameters({
+                key: FORMAT,
+                value: format,
+                currentUrlParameters: urlParameters,
+                includeMimeType: true,
+            });
+
+            urlParameters = urlParameters.concat(url);
+            returnedMimeType = mimeType;
+        }
+
+        if (resize) {
+            const url = buildUrlParameters({
+                key: RESIZE,
+                value: resize,
+                currentUrlParameters: urlParameters,
+            });
+
+            urlParameters = urlParameters.concat(url);
+        }
+
+        if (cropRadius) {
+            const url = buildUrlParameters({
+                key: CROP,
+                value: cropRadius,
+                currentUrlParameters: urlParameters,
+            });
+
+            urlParameters = urlParameters.concat(url);
+        }
+
+        if (quality) {
+            const url = buildUrlParameters({
+                key: QUALITY,
+                value: quality,
+                currentUrlParameters: urlParameters,
+            });
+
+            urlParameters = urlParameters.concat(url);
+        }
+
+        if (backgroundColor) {
+            const url = buildUrlParameters({
+                key: BACKGROUND_COLOR,
+                value: backgroundColor,
+                currentUrlParameters: urlParameters,
+            });
+
+            urlParameters = urlParameters.concat(url);
+        }
+
+        return {
+            convertedUrl: originalUrl.concat(urlParameters),
+            mimeType: returnedMimeType,
+            originalUrl,
+        };
+    }, [imageProps.src, format, resize, optimize, cropRadius, quality, backgroundColor]);
+
+    return (
+        <picture data-testid="picture">
+            { convertedUrl !== originalUrl && <source srcSet={ convertedUrl } type={ mimeType } /> }
+            <source srcSet={ originalUrl } />
+            <img { ...imageProps } ref={ ref } />
```

Comment:
Shouldn't we spread the `imageProps` to the sources too? This way only the last fallback will get those props.

Reply:
Nop, that's not how it works. 

The [documentation](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/picture) for `<picture>` says

> The <img> element serves two purposes:
> - It describes the size and other attributes of the image and its presentation.
> - It provides a fallback in case none of the offered <source> elements are able to provide a usable image.

Thus, you must use `<img>` tag to use the attributes you want to provide to your image. Furthermore, you can see [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/source) the attributes that a `<source>` element can have. For example, you're not allowed to give an `alt` attribute to a `<source>` element.

Reply:
cc @dominguesgm

Reply:
Great then, I didn't know it worked that way 🌟👌